collect_app/src/test/java/org/odk/collect/android/utilities/ArrayUtilsTest.java

```diff
@@ -0,0 +1,57 @@
+package org.odk.collect.android.utilities;
```

Comment:
Thanks for continuing to improve this. I think I should have given you better/more clear advice earlier. We can really simplify and improve this class.

As a learning exercise, you might try to find the reasons for all of my modifications here:

![screen shot 2018-03-18 at 2 17 04 pm](https://user-images.githubusercontent.com/55099/37571227-16a1131e-2ab7-11e8-990a-877ea21845de.png)

Here are the changes if you want to cherry pick the commit as is: https://github.com/dcbriccetti/collect/commit/4b9dcc86828a6ae4cea7537238b04fdaba0237d3